Test failures detected — 2026-03-17

## Test Failures Detected

**File**: `packages/policy/tests/policy-packs.test.ts`
**Package**: `@red-codes/policy`
**Detected**: 2026-03-17T12:28:28Z
**HEAD**: `6edfa9a28a907683cba447cbcd17ef9a1ba65d0c`

---

## Failing Tests (7)

All failures share the pattern `AssertionError: expected false to be true` — the policy packs are **denying actions they should allow**.

### `soc2 pack` (2 failures)

```
✗ allows push to feature branch when tests pass
  AssertionError: expected false to be true

✗ allows file writes to safe paths (within blast radius)
  AssertionError: expected false to be true
```

### `hipaa pack` (2 failures)

```
✗ allows push to feature branch
  AssertionError: expected false to be true

✗ allows file writes to safe paths (within blast radius)
  AssertionError: expected false to be true
```

### `engineering-standards pack` (3 failures)

```
✗ allows push to feature branch when tests and format pass
  AssertionError: expected false to be true

✗ allows file writes to safe paths (within blast radius)
  AssertionError: expected false to be true

✗ allows file deletion within limits
  AssertionError: expected false to be true
```

---

## Root Cause Hypothesis

The `soc2`, `hipaa`, and `engineering-standards` policy packs were added in v2.0.0. These packs are over-denying operations that should be allowed. Likely causes:

1. The new policy packs contain overly broad `deny` rules that unintentionally catch feature branch pushes and safe file operations
2. A recent change to the policy evaluator altered rule precedence (deny before allow, or missing `allow` rules)
3. The packs are missing explicit `allow` rules that the tests expect

**Note**: The `strict`, `open-source`, `enterprise`, and `ci-safe` packs all pass their tests. Only the 3 new packs (soc2, hipaa, engineering-standards) are failing.

---

## Steps to Reproduce

```bash
cd packages/policy
npx vitest run tests/policy-packs.test.ts 2>&1 | grep -E "FAIL|×"
```

---

> _Reported by Test Agent — do not close unless failures are resolved._

--- a/packages/policy/tests/policy-packs.test.ts
+++ b/packages/policy/tests/policy-packs.test.ts
@@ -1,12 +1,14 @@
 // Tests for prebuilt policy packs — verifies all seven packs load correctly,
 // have valid structure, and produce expected governance decisions.
 import { describe, it, expect } from 'vitest';
-import { resolve } from 'node:path';
+import { dirname, resolve } from 'node:path';
+import { fileURLToPath } from 'node:url';
 import { resolvePackPath, loadPackFile } from '@red-codes/policy';
 import { evaluate } from '@red-codes/policy';
 import type { LoadedPolicy, NormalizedIntent } from '@red-codes/policy';
 
-const POLICIES_DIR = resolve('../../policies');
+const __dirname = dirname(fileURLToPath(import.meta.url));
+const POLICIES_DIR = resolve(__dirname, '../../../policies');
 
 /** Helper to load a pack from the policies directory */
 function loadPack(name: string): LoadedPolicy {
@@ -493,7 +495,10 @@ describe('hipaa pack — governance decisions', () => {
   });
 
   it('denies file deletion (164.312(c)(1) — data integrity)', () => {
-    const result = evaluate(intent({ action: 'file.delete', target: 'data/records.csv' }), policies);
+    const result = evaluate(
+      intent({ action: 'file.delete', target: 'data/records.csv' }),
+      policies
+    );
     expect(result.allowed).toBe(false);
   });
 
@@ -513,10 +518,7 @@ describe('hipaa pack — governance decisions', () => {
   });
 
   it('allows push to feature branch', () => {
-    const result = evaluate(
-      intent({ action: 'git.push', branch: 'feature/hipaa-fix' }),
-      policies
-    );
+    const result = evaluate(intent({ action: 'git.push', branch: 'feature/hipaa-fix' }), policies);
     expect(result.allowed).toBe(true);
   });
 
@@ -646,7 +648,7 @@ describe('engineering-standards pack — governance decisions', () => {
 
 describe('policy packs — extends integration', () => {
   it('can be loaded as extends references from project root', () => {
-    const projectRoot = resolve('../..');
+    const projectRoot = resolve(__dirname, '../../..');
     const packPath = resolvePackPath('./policies/strict', projectRoot);
     expect(packPath).not.toBeNull();
     const pack = loadPackFile(packPath!);
@@ -687,7 +689,7 @@ describe('policy packs — extends integration', () => {
   it.each(['soc2', 'hipaa', 'engineering-standards'])(
     'compliance pack %s can be loaded from project root',
     (name) => {
-      const projectRoot = resolve('../..');
+      const projectRoot = resolve(__dirname, '../../..');
       const packPath = resolvePackPath(`./policies/${name}`, projectRoot);
       expect(packPath).not.toBeNull();
       const pack = loadPackFile(packPath!);

